gfauto/gfauto/amber_converter.py

```diff
@@ -513,51 +513,62 @@ def get_amber_script_shader_def(shader: Shader, name: str) -> str:
 def graphics_shader_job_amber_test_to_amber_script(
     shader_job_amber_test: ShaderJobBasedAmberTest, amberfy_settings: AmberfySettings
 ) -> str:
-    # TODO: handle reference, if present.
-
     # Guaranteed, and needed for type checker.
     assert isinstance(shader_job_amber_test.variant, GraphicsShaderJob)  # noqa
 
     result = get_amber_script_header(amberfy_settings)
 
-    variant = shader_job_amber_test.variant
+    jobs = [shader_job_amber_test.variant]
 
-    variant_vertex_shader_name = f"{variant.name_prefix}_vertex_shader"
-    variant_fragment_shader_name = f"{variant.name_prefix}_fragment_shader"
+    if shader_job_amber_test.reference:
+        jobs.append(shader_job_amber_test.reference)
```

Comment:
Annoyingly, you will need to add: `assert isinstance(shader_job_amber_test.reference, GraphicsShaderJob)  # noqa`

The # noqa is needed because assertions are generally banned from non-test code, but it is only thing the type checker recognizes.